commit ed654c527297b5730de1503f55998454505eec10
Author: Kirill A. Korinsky <kirill@korins.ky>
Date:   Tue Sep 21 19:41:09 2021 +0200

    Allow to control resources path
    
    This commit introduced two CMake options `OGRE_MEDIA_PATH` and
    `OGRE_PLUGINS_PATH` to control such paths.
    
    Also, it keeps `OGRE_CFG_INSTALL_PATH`, `OGRE_CMAKE_DIR` and
    `OGRE_PLUGIN_DIR_CMAKE` untouched when it exist.

diff --git a/CMake/InstallResources.cmake b/CMake/InstallResources.cmake
index 9daa3dfb3..8211367f0 100644
--- a/CMake/InstallResources.cmake
+++ b/CMake/InstallResources.cmake
@@ -11,33 +11,63 @@
 # Generate and install the config files needed for the samples
 ##################################################################
 
-if (WIN32)
-  set(OGRE_MEDIA_PATH "Media")
+if (NOT OGRE_MEDIA_PATH)
+  if (WIN32)
+    set(OGRE_MEDIA_PATH "Media")
+    set(OGRE_MEDIA_DIR_REL "${CMAKE_INSTALL_PREFIX}/${OGRE_MEDIA_PATH}")
+  elseif (APPLE)
+    set(OGRE_MEDIA_PATH "Media")
+    if(APPLE_IOS)
+      set(OGRE_MEDIA_DIR_REL "${OGRE_MEDIA_PATH}")
+    else()
+      set(OGRE_MEDIA_DIR_REL "../${OGRE_MEDIA_PATH}")
+    endif()
+  elseif (UNIX)
+    set(OGRE_MEDIA_PATH "share/OGRE/Media")
+    set(OGRE_MEDIA_DIR_REL "./Media")
+  endif()
+else ()
   set(OGRE_MEDIA_DIR_REL "${CMAKE_INSTALL_PREFIX}/${OGRE_MEDIA_PATH}")
+endif()
+
+if (NOT OGRE_PLUGINS_PATH)
+  if (WIN32)
+    set(OGRE_PLUGINS_PATH "bin")
+    set(OGRE_PLUGIN_DIR_REL "${CMAKE_INSTALL_PREFIX}/bin")
+  elseif (APPLE)
+    set(OGRE_PLUGINS_PATH "lib/${CMAKE_BUILD_TYPE}")
+    set(OGRE_PLUGIN_DIR_REL "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_BUILD_TYPE}/")
+  elseif (UNIX)
+    set(OGRE_PLUGINS_PATH "${OGRE_LIB_DIRECTORY}/OGRE")
+    set(OGRE_PLUGIN_DIR_REL "${CMAKE_INSTALL_PREFIX}/${OGRE_LIB_DIRECTORY}/OGRE")
+  endif()
+else ()
+  set(OGRE_PLUGIN_DIR_REL "${CMAKE_INSTALL_PREFIX}/${OGRE_PLUGINS_PATH}")
+endif()
+
+if (NOT OGRE_CFG_INSTALL_PATH)
+  if (WIN32 OR APPLE)
+    set(OGRE_CFG_INSTALL_PATH "bin")
+  elseif (UNIX)
+    set(OGRE_CFG_INSTALL_PATH "share/OGRE")
+  endif()
+endif()
+
+if (WIN32)
   set(OGRE_TEST_MEDIA_DIR_REL "../Tests/${OGRE_MEDIA_PATH}")
-  set(OGRE_PLUGIN_DIR_REL "${CMAKE_INSTALL_PREFIX}/bin")
   set(OGRE_SAMPLES_DIR_REL ".")
-  set(OGRE_CFG_INSTALL_PATH "bin")
 elseif (APPLE)
-  set(OGRE_MEDIA_PATH "Media")
   if(APPLE_IOS)
-    set(OGRE_MEDIA_DIR_REL "${OGRE_MEDIA_PATH}")
     set(OGRE_TEST_MEDIA_DIR_REL "../../Tests/${OGRE_MEDIA_PATH}")
   else()
-    set(OGRE_MEDIA_DIR_REL "../${OGRE_MEDIA_PATH}")
     set(OGRE_TEST_MEDIA_DIR_REL "${CMAKE_INSTALL_PREFIX}/Tests/Media")
   endif()
   # these are resolved relative to the app bundle
-  set(OGRE_PLUGIN_DIR_REL "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_BUILD_TYPE}/")
   set(OGRE_SAMPLES_DIR_REL "Contents/Plugins/")
   set(OGRE_CFG_INSTALL_PATH "bin")
 elseif (UNIX)
-  set(OGRE_MEDIA_PATH "share/OGRE/Media")
-  set(OGRE_MEDIA_DIR_REL "./Media")
   set(OGRE_TEST_MEDIA_DIR_REL "${CMAKE_INSTALL_PREFIX}/Tests/Media")
-  set(OGRE_PLUGIN_DIR_REL "${CMAKE_INSTALL_PREFIX}/${OGRE_LIB_DIRECTORY}/OGRE")
   set(OGRE_SAMPLES_DIR_REL "${CMAKE_INSTALL_PREFIX}/${OGRE_LIB_DIRECTORY}/OGRE/Samples")
-  set(OGRE_CFG_INSTALL_PATH "share/OGRE")
 endif ()
 
 # generate OgreConfigPaths.h
@@ -210,16 +240,22 @@ endif ()
 # Create the CMake package files
 include(CMakePackageConfigHelpers)
 
-if(WIN32 OR APPLE)
-  set(OGRE_CMAKE_DIR "CMake")
-else()
-  set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/OGRE/cmake")
+if (NOT OGRE_CMAKE_DIR)
+  if(WIN32 OR APPLE)
+    set(OGRE_CMAKE_DIR "CMake")
+  else()
+    set(OGRE_CMAKE_DIR "${OGRE_LIB_DIRECTORY}/OGRE/cmake")
+  endif()
 endif()
-if(WIN32)
-  set(OGRE_PLUGIN_DIR_CMAKE "bin")
-else()
-  set(OGRE_PLUGIN_DIR_CMAKE "${OGRE_LIB_DIRECTORY}/OGRE")
+
+if (NOT OGRE_PLUGIN_DIR_CMAKE)
+  if(WIN32)
+    set(OGRE_PLUGIN_DIR_CMAKE "bin")
+  else()
+    set(OGRE_PLUGIN_DIR_CMAKE "${OGRE_LIB_DIRECTORY}/OGRE")
+  endif()
 endif()
+
 configure_package_config_file(${OGRE_TEMPLATES_DIR}/OGREConfig.cmake.in ${PROJECT_BINARY_DIR}/cmake/OGREConfig.cmake
     INSTALL_DESTINATION ${OGRE_CMAKE_DIR}
     PATH_VARS OGRE_MEDIA_PATH OGRE_PLUGIN_DIR_CMAKE OGRE_CFG_INSTALL_PATH CMAKE_INSTALL_PREFIX)
